src/app.js

```diff
@@ -0,0 +1,21 @@
+import express from "express";
+import connectsToDb from "./config/dbConnect.js";
+import routes from "./routes/index.js";
+import errorHandler from "./middlewares/errorHandler.js";
+
+const app = express();
+routes(app);
+
+const connection = await connectsToDb();
+
+connection.on("error", (err) => {
+  console.error("Não foi possível se conectar ao banco de dados:", err);
+});
+
+connection.once("open", () => {
+  console.log("Conexão com o banco de dados feita com sucesso!");
+});

```

Comment:
O ideal aqui é que essas definições dos listeners de status de conexão do banco de dados sejam feitas no arquivo de configuração do MongoDB (nesse caso o arquivo `dbConnect`), por dois motivos:

1. A gente centraliza tudo que é relacionado à conexão de banco de dados do MongoDB dentro do próprio arquivo de configuração e instanciação dele. A vantagem disso é que a gente mantém tudo o que é relacionado a MongoDB dentro do seu próprio arquivo e não corre o risco de perder o controle dessas informações no código.

2. Nesse caso em específico, nós só temos o MongoDB. Mas pensando em um contexto macro, onde a API se conecta com outros bancos (MariaDB, Redis, etc.), já pensou se a gente joga todas essas configurações aqui no app? Ele ficaria enorme e lidando com responsabilidades que não é dele!

Uma sugestão: no arquivo `dbConnect.js`, tu podes ter um método `init()`. Esse método pode ter a seguinte implementação:

```javascript
async function init() {
  const connection = await connectsToDB();

  connection.on("error", (err) => {
    console.error("Não foi possível se conectar ao banco de dados:", err);
  });

  connection.once("open", () => {
    console.log("Conexão com o banco de dados feita com sucesso!");
  });
}

export default { init };
```

E daí nesse arquivo `app.js`, tu pode fazer o seguinte:

```js
...,
import mongoDB from './config/dbConnect.js'
...

await mongoDB.init();
```

Daí tu centralizada tudo quanto for relacionado à configuração do MongoDB dentro de um único arquivo, e só chama o `init()` para iniciar a conexão e definir os listeners. O que acha?

Reply:
Com toda certeza!

Tá feito! :)